### PR TITLE
TE-1066 Adapt Summary component for mobile

### PR DESCRIPTION
**Pull request by red-runner**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1066)

### What **one** thing does this PR do?
Adapts the `Summary` component to mobile resolution
### Any other notes
summary

### CODE REVIEW

File: src/components/property-page-widgets/Summary/component.js [hunk 1]
```diff
@@ -17,15 +19,29 @@ export const Component = ({
   ratingNumber,
 }) => (
   <Segment.Group compact>
-    <Segment>
-      <Heading>{propertyName}</Heading>
-    </Segment>
-    <Segment.Group horizontal>
+    <ShowOnDesktop parent="Fragment">
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/component.js]: I don't think this `parent="Fragment"` is doing anything. If you want the parent to be a `Fragment` then `Fragment` needs to be imported and `<ShowOnDesktop parent={Fragment}>`

File: src/components/property-page-widgets/Summary/component.js [hunk 2]
```diff
@@ -17,15 +19,29 @@ export const Component = ({
...
+            disabled
+            maxRating={5}
+            rating={Math.round(ratingNumber)}
+            size="tiny"
+          />
+        </Segment>
+        <Segment>{getNightPriceMarkup(nightPrice, 'small')}</Segment>
+      </Segment.Group>
+    </ShowOnDesktop>
+    <ShowOnMobile parent="Fragment">
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/component.js]: same as: https://github.com/lodgify/lodgify-ui/pull/348/files#r218735777

File: src/components/property-page-widgets/Summary/component.js [hunk 3]
```diff
@@ -34,8 +50,8 @@ export const Component = ({
           size="tiny"
         />
       </Segment>
-      <Segment>{getNightPriceMarkup(nightPrice, 'small')}</Segment>
-    </Segment.Group>
+      <Segment as="p">{getNightPriceMarkup(nightPrice, 'small')}</Segment>
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/component.js]: `as="p"` is not necessary

File: src/components/property-page-widgets/Summary/component.js [hunk 4]
```diff
@@ -17,15 +19,29 @@ export const Component = ({
...
+            maxRating={5}
+            rating={Math.round(ratingNumber)}
+            size="tiny"
+          />
+        </Segment>
+        <Segment>{getNightPriceMarkup(nightPrice, 'small')}</Segment>
+      </Segment.Group>
+    </ShowOnDesktop>
+    <ShowOnMobile parent="Fragment">
+      <Segment as="p">
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/component.js]: as="p" is not necessary

File: src/components/property-page-widgets/Summary/component.js (hunk 3 above)
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/component.js]: Lines 44 to 55 are duplicate so you can pull this out into a function that returns this markup `getRatingAndNightPriceMarkup`

File: src/components/property-page-widgets/Summary/component.js (hunk 1 above)
Comment (red-runner) [reply src/components/property-page-widgets/Summary/component.js]: @Jonnygwi This was wrapped in `<Fragment>` before. I'm not specially fond of it, so I guess we could get rid of it. Thoughts?

File: src/components/property-page-widgets/Summary/utils/getHeading.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+ * @param  {string} propertyName
+ * @return {Object}
+ */
+export const getHeading = propertyName => {
+  return (
+    <Segment>
+      <Heading>{propertyName}</Heading>
+    </Segment>
+  );
+};
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/utils/getHeading.js]: You can drop the curlies in place of an implicit return

```
export const getHeading = propertyName => (
    <Segment>
      <Heading>{propertyName}</Heading>
    </Segment>
  );
```

File: src/components/property-page-widgets/Summary/utils/getRatingAndNightPriceMarkup.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+          disabled
+          maxRating={5}
+          rating={Math.round(ratingNumber)}
+          size="tiny"
+        />
+      </Segment>
+      <Segment>{getNightPriceMarkup(nightPrice, 'small')}</Segment>
+    </Segment.Group>
+  );
+};
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/utils/getRatingAndNightPriceMarkup.js]:
```
export const getRatingAndNightPriceMarkup = (ratingNumber, nightPrice) => {
  return (
    <Segment.Group horizontal>
      <Segment>
        {ratingNumber}
        <Rating
          disabled
          maxRating={5}
          rating={Math.round(ratingNumber)}
          size="tiny"
        />
      </Segment>
      <Segment>{getNightPriceMarkup(nightPrice, 'small')}</Segment>
    </Segment.Group>
  );
};
```

File: src/components/property-page-widgets/Summary/utils/getHeading.spec.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+import { getHeading } from './getHeading';
+
+const propertyName = 'USS-Enterprise';
+
+const getMarkupAsRenderedComponent = () =>
+  shallow(<div>{getHeading(propertyName)}</div>).children();
+
+describe('getHeading', () => {
+  it('render a single `Segment`', () => {
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/utils/getHeading.spec.js]: `'should render a single `Segment`'`

File: src/components/property-page-widgets/Summary/utils/getLocationName.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+ * @return {Object}
+ */
+export const getLocationName = locationName => {
+  return (
+    <Segment>
+      {locationName}
+      <Icon color="yellow" name={ICON_NAMES.MAP_PIN} size="small" />
+    </Segment>
+  );
+};
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/utils/getLocationName.js]: same here: https://github.com/lodgify/lodgify-ui/pull/348/files#r219112175

File: src/components/property-page-widgets/Summary/utils/getHeading.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { Segment } from 'semantic-ui-react';
+
+import { Heading } from 'typography/Heading';
+
+/**
+ * @param  {string} propertyName
+ * @return {Object}
+ */
+export const getHeading = propertyName => {
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/utils/getHeading.js]: `getHeading` -> `getHeadingMarkup`

File: src/components/property-page-widgets/Summary/utils/getLocationName.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Segment } from 'semantic-ui-react';
+
+import { Icon, ICON_NAMES } from 'elements/Icon';
+
+/**
+ * @param  {string} locationName
+ * @return {Object}
+ */
+export const getLocationName = locationName => {
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/utils/getLocationName.js]: `getLocationName` -> `getLocationNameMarkup`

File: src/components/property-page-widgets/Summary/utils/getRatingAndNightPriceMarkup.spec.js [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    const wrapper = getMarkupAsRenderedComponent();
+
+    expectComponentToBe(wrapper, Segment.Group);
+  });
+  it('should have the right props', () => {
+    const wrapper = getMarkupAsRenderedComponent().find(Segment.Group);
+
+    const actual = wrapper.prop('horizontal');
+
+    expect(actual).toBe(true);
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/utils/getRatingAndNightPriceMarkup.spec.js]: use `expectComponentToHaveProps`

File: src/components/property-page-widgets/Summary/utils/getRatingAndNightPriceMarkup.spec.js [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+describe('getNightPriceMarkup', () => {
+  it('should render a single `Segment.Group` component', () => {
+    const wrapper = getMarkupAsRenderedComponent();
+
+    expectComponentToBe(wrapper, Segment.Group);
+  });
+  it('should have the right props', () => {
+    const wrapper = getMarkupAsRenderedComponent().find(Segment.Group);
+
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/utils/getRatingAndNightPriceMarkup.spec.js]: Remove this empty line

File: src/components/property-page-widgets/Summary/component.spec.js [hunk 13]
```diff
@@ -23,171 +20,62 @@ const props = {
...
+          .at(1);
 
-    it('should have the right props', () => {
-      const wrapper = getFirstIcon();
-
-      expectComponentToHaveProps(wrapper, {
-        color: 'yellow',
-        name: 'map pin',
-        size: 'small',
+        expectComponentToHaveProps(wrapper, { horizontal: true });
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/component.spec.js]: this should be in a different `it` block as it's checking for props not children

File: src/components/property-page-widgets/Summary/component.spec.js [hunk 14]
```diff
@@ -23,171 +20,62 @@ const props = {
...
+      expectComponentToHaveChildren(wrapper, Segment, Segment.Group);
     });
-  });
 
-  describe('the first `Icon` component', () => {
-    const getFirstIcon = () =>
-      getSummary()
-        .find(Icon)
-        .at(0);
+    describe('if props.isUserMobile is true', () => {
```
Comment (Jonnygwi) [inline src/components/property-page-widgets/Summary/component.spec.js]: `if props.isUserMobile is true` -> `if `props.isUserOnMobile` is true`

**Pull request merged**